crates/noirc_evaluator/src/ssa_refactor/ir/basic_block.rs

```diff
@@ -92,7 +97,7 @@ impl BasicBlock {
     /// Removes the given instruction from this block if present or panics otherwise.
     pub(crate) fn remove_instruction(&mut self, instruction: InstructionId) {
         let index =
-            self.instructions.iter().position(|id| *id == instruction).unwrap_or_else(|| {
+            self.instructions.iter().rev().position(|id| *id == instruction).unwrap_or_else(|| {
```

Comment:
Please comment why the search is from the end backwards

Reply:
Sure thing. It was just an optimization after the observation that we usually call this on instructions at the end of a block rather than the beginning.